packages/grafana-ui/src/components/Table/FilterPopup.tsx

```diff
@@ -0,0 +1,159 @@
+
+  const onFilter = useCallback(
+    (event: React.MouseEvent) => {
+      const filtered = values.length ? values : undefined;
+
+      setFilter(filtered);
+      onClose();
+    },
+    [setFilter, values, onClose]
+  );
+
+  const onClearFilter = useCallback(
+    (event: React.MouseEvent) => {
+      setFilter(undefined);
+      onClose();
+    },
+    [setFilter, onClose]
+  );
+
+  const clearFilterVisible = useMemo(() => filterValue !== undefined, [filterValue]);
+
+  const theme = useTheme();
+  const styles = getStyles(theme);
+
+  return (
+    <ClickOutsideWrapper onClick={onCancel} useCapture={true}>
+      <div className={cx(styles.filterContainer)} onClick={stopPropagation}>
+        <VerticalGroup spacing="lg">
+          <VerticalGroup spacing="xs">
+            <Label>Filter by values:</Label>
+            <div className={cx(styles.listDivider)} />
+            <FilterList onChange={setValues} values={values} options={options} />
+          </VerticalGroup>
+          <HorizontalGroup spacing="lg">
+            <HorizontalGroup>
+              <Button size="sm" onClick={onFilter}>
+                Ok
+              </Button>
+              <Button size="sm" variant="secondary" onClick={onCancel}>
+                Cancel
+              </Button>
+            </HorizontalGroup>
+            {clearFilterVisible && (
+              <HorizontalGroup>
+                <Button variant="link" size="sm" onClick={onClearFilter}>
+                  Clear filter
+                </Button>
+              </HorizontalGroup>
+            )}
+          </HorizontalGroup>
+        </VerticalGroup>
+      </div>
+    </ClickOutsideWrapper>
+  );
+};
+
+const getStyles = stylesFactory((theme: GrafanaTheme) => ({
+  filterContainer: css`
+    label: filterContainer;
+    width: 100%;
+    min-width: 250px;
+    height: 100%;
+    max-height: 400px;
+    background-color: ${theme.colors.bg1};
+    border: ${theme.border.width.sm} solid ${theme.colors.border2};
+    padding: ${theme.spacing.md};
+    margin: ${theme.spacing.sm} 0;
+    box-shadow: 0px 0px 20px ${theme.colors.dropdownShadow};
+    border-radius: ${theme.spacing.xs};
+  `,
+  listDivider: css`
+    label: listDivider;
+    width: 100%;
+    border-top: ${theme.border.width.sm} solid ${theme.colors.border2};
+    padding: ${theme.spacing.xs} ${theme.spacing.md};
+  `,
+}));
+
+const calculateUniqueFieldValues = (rows: any[], field?: Field) => {
+  if (!field || rows.length === 0) {
+    return {};
+  }
+
+  const set: Record<string, any> = {};
+
+  for (let index = 0; index < rows.length; index++) {
+    const fieldIndex = parseInt(rows[index].id, 10);
+    const fieldValue = field.values.get(fieldIndex);
+    const displayValue = field.display ? field.display(fieldValue) : fieldValue;
+    const value = field.display ? formattedValueToString(displayValue) : displayValue;
+    set[value || '(Blanks)'] = fieldValue;
+  }
+
+  return set;
+};
```

Comment:
Maybe we could add some tests for these utility functions (and even move them out to a separate file like `FilterPopup.utils.tsx`)?

Reply:
Yes, great suggestion. Will look into that!

Reply:
Done